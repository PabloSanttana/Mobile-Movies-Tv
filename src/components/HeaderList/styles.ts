import styled from "styled-components/native";
import { convertScale, sizeDeviceTypeScale } from "@src/utils/utils";
import { DeviceTypeProps } from "@src/interfaces";

type DeviceType = {
  deviceType: DeviceTypeProps;
};

export const Container = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin: 0px ${(props) => props.theme.space.marginHorizontal + "px"};
  margin-top: ${convertScale(25)};
  margin-bottom: ${convertScale(15)};
`;

export const Title = styled.Text<DeviceType>`
  font-family: ${(props) => props.theme.fonts.title};
  font-size: ${(props) =>
    sizeDeviceTypeScale(props.deviceType, 14, props.theme.size.big)};
  color: ${(props) => props.theme.colors.textSession};
`;

export const Button = styled.TouchableOpacity`
  border: ${convertScale(1)} solid
    ${(props) => props.theme.colors.textSecondary};
  padding: ${convertScale(2)} ${convertScale(10)};
  border-radius: ${convertScale(10)};
`;

export const ButtonText = styled.Text<DeviceType>`
  font-family: ${(props) => props.theme.fonts.subtitle};
  font-size: ${(props) =>
    sizeDeviceTypeScale(props.deviceType, 8, props.theme.size.small)};
  color: ${(props) => props.theme.colors.textSecondary};
`;
