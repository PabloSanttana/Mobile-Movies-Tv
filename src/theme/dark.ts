export const dark = {
  colors: {
    backgroundPrimary: "#0F0F0F",
    backgroundSecondary: "#242425",
    star: "#F2A33A",
    red: "#CF353F",
    textPrimary: "#fff",
    textSecondary: "#c5c5c5",

    textCategory: "#000",
    backgroundCategory: "#F2A33A",

    routerIcon: "#DBDBDB",
    routerIconActive: "#F2A33A",
    textSession: "#F2A33A",
  },
  size: {
    big: 18,
    average: 15,
    small: 12,
  },
  fonts: {
    title: "Roboto_700Bold",
    subtitle: "Roboto_400Regular",
    text: "Roboto_300Light",
  },
  space: {
    marginHorizontal: 20,
  },
};
