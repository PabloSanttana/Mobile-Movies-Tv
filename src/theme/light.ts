export const light = {
  colors: {
    backgroundPrimary: "#EFF1F2",
    backgroundSecondary: "#FFFFFF",
    star: "#F2A33A",
    red: "#e9140a",
    textPrimary: "#000",
    textSecondary: "#5A5A5A",
    textCategory: "#304b8c",
    backgroundCategory: "#DCE3FF",
    routerIcon: "#BDBCCE",
    routerIconActive: "#110E47",
    textSession: "#110F48",
  },
  size: {
    big: 18,
    average: 13,
    small: 11,
  },
  fonts: {
    title: "Roboto_700Bold",
    subtitle: "Roboto_400Regular",
    text: "Roboto_300Light",
  },
  space: {
    marginHorizontal: 20,
  },
};
