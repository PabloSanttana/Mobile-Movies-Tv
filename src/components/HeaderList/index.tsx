import { DeviceTypeProps } from "@src/interfaces";
import React from "react";
import { Button, ButtonText, Container, Title } from "./styles";

export type HeaderListProps = {
  title: string;
  onPress: () => void;
  isMore?: boolean;
  deviceType: DeviceTypeProps;
};

function HeaderList({
  title,
  onPress,
  isMore = true,
  deviceType,
}: HeaderListProps) {
  return (
    <Container>
      <Title deviceType={deviceType}>{title}</Title>
      {isMore && (
        <Button onPress={() => onPress()} activeOpacity={0.7}>
          <ButtonText deviceType={deviceType}>Ver mais</ButtonText>
        </Button>
      )}
    </Container>
  );
}

function arePropsEqual(prevProps: HeaderListProps, nextProps: HeaderListProps) {
  if (prevProps.title === nextProps.title) {
    return true;
  }
  return false;
}

export default React.memo(HeaderList, arePropsEqual);
