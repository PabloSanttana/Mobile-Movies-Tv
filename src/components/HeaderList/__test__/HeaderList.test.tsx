import React from "react";
import { ThemeProvider } from "styled-components/native";
import { dark } from "../../../theme/dark";
import { render, fireEvent } from "@testing-library/react-native";
import HeaderList, { HeaderListProps, arePropsEqualHeaderList } from "../index";

describe("HeaderList", () => {
  describe("Compontes renders", () => {
    const defaultProps: HeaderListProps = {
      title: "Test Header",
      onPress: jest.fn(),
      deviceType: "mobile",
      isMore: true,
    };
    it("should render with title", () => {
      const { getByText } = render(
        <ThemeProvider theme={dark}>
          <HeaderList {...defaultProps} />
        </ThemeProvider>
      );
      expect(getByText("Test Header")).toBeDefined();
    });
    it("should render with title is not isMore", () => {
      const data = {
        title: "Test Header",
        onPress: jest.fn(),
        deviceType: "tablet",
      };
      const { getByText } = render(
        <ThemeProvider theme={dark}>
          <HeaderList {...data} />
        </ThemeProvider>
      );
      expect(getByText("Test Header")).toBeDefined();
    });
    it('should call onPress when "Ver mais" button is pressed', () => {
      const { getByText } = render(
        <ThemeProvider theme={dark}>
          <HeaderList {...defaultProps} />
        </ThemeProvider>
      );
      fireEvent.press(getByText("Ver mais"));
      expect(defaultProps.onPress).toHaveBeenCalledTimes(1);
    });
    it('should not render "Ver mais" button when isMore prop is false', () => {
      const { queryByText } = render(
        <ThemeProvider theme={dark}>
          <HeaderList {...defaultProps} isMore={false} />
        </ThemeProvider>
      );
      expect(queryByText("Ver mais")).toBeNull();
    });

    it("should pass correct props to Container and Title components", () => {
      const { getByTestId } = render(
        <ThemeProvider theme={dark}>
          <HeaderList {...defaultProps} />
        </ThemeProvider>
      );
      const title = getByTestId("title");

      expect(title.props.children).toBe("Test Header");
      expect(title.props.style[0].fontSize).toEqual(expect.any(Number));
    });
    it("should render memoized component", () => {
      const { rerender } = render(
        <ThemeProvider theme={dark}>
          <HeaderList {...defaultProps} />
        </ThemeProvider>
      );
      const memoizedComponent = rerender(
        <ThemeProvider theme={dark}>
          <HeaderList {...defaultProps} />
        </ThemeProvider>
      );
      expect(memoizedComponent).toMatchSnapshot();
    });
  });

  describe("arePropsEqualHeaderList function is correct", () => {
    it("returns true if previous and next props have the same title", () => {
      const prevProps = {
        title: "Header Title",
        deviceType: "phone",
        onPress: jest.fn(),
      };
      const nextProps = {
        title: "Header Title",
        deviceType: "phone",
        onPress: jest.fn(),
      };
      expect(arePropsEqualHeaderList(prevProps, nextProps)).toBe(true);
    });

    it("returns false if previous and next props have different titles", () => {
      const prevProps = {
        title: "Header Title",
        deviceType: "phone",
        onPress: jest.fn(),
      };
      const nextProps = {
        title: "New Header Title",
        deviceType: "phone",
        onPress: jest.fn(),
      };
      expect(arePropsEqualHeaderList(prevProps, nextProps)).toBe(false);
    });
  });
});
