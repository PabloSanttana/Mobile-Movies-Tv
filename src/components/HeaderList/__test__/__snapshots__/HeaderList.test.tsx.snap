// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderList Compontes renders should render memoized component 1`] = `undefined`;
